pkg/api/types.go

```diff
@@ -1205,6 +1208,21 @@ type KeyToPath struct {
 	Mode *int32
 }
 
+// LocalVolume represents a local volume accessible only from one node
+type LocalVolumeSource struct {
+	// NodeName represents the node that the volume is located at
+	NodeName string
+	// Fs represents a filesystem-based local volume
+	// +optional
+	Fs *LocalFsVolume
+}
+
+// LocalFsVolume represents a filesystem-based local volume
+type LocalFsVolume struct {
```

Comment:
> can we find a way to solve it for everyone instead of being new for this one?

Depends on at what level `block` vs `file` is defined. Ideally it has to be surfaced by the underlying storage volume and so this will require API changes to all existing volume types.